component/topology/topology.go

```diff
@@ -34,16 +34,10 @@ func GetCurrentComponent() []Component {
 }
 
 func GetEtcdClient() *clientv3.Client {
-	if discover == nil {
-		return nil
-	}
-	return discover.etcdCli
+	return discover.cli.etcdCli
```

Comment:
Please don't do this. Avoid shared globals.

Commonly, for shared globals, you will make these things worse:
1. Now it's hard to write unit tests. Not to mention running unit tests concurrently.
2. Hard to ensure concurrent safety. You left this possible to be modified and accessed EVERYWHERE in the code. This is horrible. You may need to take a look at every references to ensure that it is not modified concurrently.

Reply:
Ok, I will do some refactor in another PR, This PR is a bug fix, and there are many other global share variable.